Sources/StreamChatUI/ChatMessageList/MessageComposer/ChatMessageComposerImageAttachmentsView.swift

```diff
@@ -5,44 +5,39 @@
 import StreamChat
 import UIKit
 
+/// A view that displays a collection of image attachments
 public typealias ChatMessageComposerImageAttachmentsView = _ChatMessageComposerImageAttachmentsView<NoExtraData>
 
+/// A view that displays a collection of image attachments
 open class _ChatMessageComposerImageAttachmentsView<ExtraData: ExtraDataTypes>: _View,
     UIConfigProvider,
     UICollectionViewDelegate,
     UICollectionViewDataSource {
-    // MARK: - Properties
-    
-    open var imagePreviewItemSize: CGSize = .init(width: 100, height: 100)
-    
-    public var images: [UIImage] = [] {
+    /// The image attachment item size. Can be overridden in `setUpLayout()`.
+    private var imagePreviewItemSize: CGSize = .init(width: 100, height: 100)
```

Comment:
If I understand it correctly, the height of the inner collection view is the same as the height of this component, right? This also limits the max height of the cell while the width of the cell should (eventually) be defined by the ratio of the attachment preview.

Seems to me that this view shouldn't have any inners sizing information and it's height can be set from the outside by its superview :)

Reply:
Just to fill in, this size is needed for `flowLayout.itemSize` to setup the size in the `collectionView` otherwise this just doesn't resize properly and we cannot see the cells... Also if this stays, I would definitely make it public and somehow highlight that this value basically defines height of the view... 

The last sentence feels a bit bad attitude, so maybe reconsider some bigger change as Vojta suggested.

Reply:
Yes like @DominikBucher12 this size is to ser the `collectionView.itemSize`. I also reuse it to set the collection view height. But you are right, the collection view can have constraints to the super view, and then the height is defined by the superview, which makes more sense 👍